src/components/forms/PlaceOrderForm.tsx

```diff
@@ -37,67 +38,93 @@ export const PlaceOrderForm: React.FC<PlaceOrderFormProps> = ({
   selectedColor,
 }) => {
   const [isSubmitted, setIsSubmitted] = useState(false);
+  const [isError, setIsError] = useState(false);
+
   const form = useForm<z.infer<typeof FormSchema>>({
     resolver: zodResolver(FormSchema),
   });
 
-  const onSubmit = (data: z.infer<typeof FormSchema>) => {
-    console.log({ brandName, modelName, selectedColor, ...data });
-    setIsSubmitted(true);
-  };
+  const onSubmit = async (data: z.infer<typeof FormSchema>) => {
+    try {
+      // simulating backend request which fails half the time
+      await new Promise((resolve, reject) => {
+        setTimeout(() => {
+          const random = Math.random();
+          if (random > 0.5) {
+            resolve(null);
+          } else {
+            reject(new Error('Simulated backend error'));
+          }
+        }, 1000);
+      });
```

Comment:
submitting the form in a promise should make it easier to connect to the backend later on (I think)